### PR TITLE
Hot 1959

### PR DESCRIPTION
**Pull request by jimlebeau**
- HOT-1959 - Save CSEC screening data to CWS/CMS.
- Functional test for /authorize/client{id} end point.
- Refactor RelationshipService.findForIds() to authorize client access before returning relationships.

## Description

**HOT-1959**
Save concerns about CSEC from a CARES screening that is submitted to CWS/CMS, so that an investigator can assess CSEC concerns about minors named on the investigation.  This story is for the back-end work only.

**Functional test for GET /authorize/client{id}**
Test that will verify that a logged in user has authorization to access client information using this end point.  This end-point uses the gov.ca.cwds.rest.services.auth.AuthorizationService.java service to return a status of 200 when authorized (or client does not exist) and a status of 403 when not authorized.

**RelationshipService.findForIds()**
This method was refactored so that access authorization is checked before the relationship information is returned from the service.  This service is currently used by the GET /clients/relationships end point.  Previously, client relationship information would always be returned regardless of the users access privilege to the client.

## Jira Issue link

https://osi-cwds.atlassian.net/browse/HOT-1959

## Tests
- [X] I have included unit tests 
- [X] I have included functional tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [X] New feature (non-breaking change which adds functionality)
- [X] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the google code style of this project.
- [X] I have ran all tests for the project.
- [X] My code is in a stable state ready to be deployable, but not necessarily complete.
- [X] I promise on my honor as a CWDS developer to ensure I don't break the build, to check SonarQube after the build completes, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: src/main/java/gov/ca/cwds/rest/services/RelationshipsService.java [hunk 1]
```diff
@@ -62,13 +66,14 @@ public Response find(String id) {
...
    */
-  @SuppressWarnings("unchecked")
   public Response findForIds(List<String> clientIds) {
-    final Set relationships = new HashSet<>();
-    if (clientIds != null){
+    final Set<Relationship> relationships = new HashSet<Relationship>();
+    if (clientIds != null) {
+      
       for (String id : clientIds) {
         if (authorized(id)) {
```
Comment (SergiiRedchuk) [inline src/main/java/gov/ca/cwds/rest/services/RelationshipsService.java]: using authorizationService.filterClientIds(...) to filter all authorized client ID's at once would be better for performance

File: src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInLakeCountyWithSealedAccess.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import org.junit.Test;
+import static io.restassured.RestAssured.given;
+import static org.junit.Assert.assertFalse;
+import gov.ca.cwds.api.FunctionalTest;
+import gov.ca.cwds.api.builder.HttpRequestHandler;
+import gov.ca.cwds.rest.authenticate.UserGroup;
+import gov.ca.cwds.rest.core.Api;
+import io.restassured.http.ContentType;
+
+public class ClientAccessForStaffInLakeCountyWithSealedAccess extends FunctionalTest {
```
Comment (SergiiRedchuk) [inline src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInLakeCountyWithSealedAccess.java]: the class name should have been ended with "Test" suffix

File: src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInLakeCountyWithSensitiveAccess.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+ * 
+ * Functional test for client access (/authorize) when the staff person (user)
+ * has sensitive access and is in Lake county.
+ * 
+ * Reference Section 4 of the Security_V2.7 document (Table 4 -  Client Search Access).
+ * 
+ * @author CWDS API Team
+ *
+ */
+public class ClientAccessForStaffInLakeCountyWithSensitiveAccess extends FunctionalTest {
```
Comment (SergiiRedchuk) [inline src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInLakeCountyWithSensitiveAccess.java]: the class name should have been ended with "Test" suffix

File: src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInStateWithSealedAccess.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import org.junit.Test;
+import static io.restassured.RestAssured.given;
+import static org.junit.Assert.assertFalse;
+import gov.ca.cwds.api.FunctionalTest;
+import gov.ca.cwds.api.builder.HttpRequestHandler;
+import gov.ca.cwds.rest.authenticate.UserGroup;
+import gov.ca.cwds.rest.core.Api;
+import io.restassured.http.ContentType;
+
+public class ClientAccessForStaffInStateWithSealedAccess extends FunctionalTest {
```
Comment (SergiiRedchuk) [inline src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInStateWithSealedAccess.java]: the class name should have been ended with "Test" suffix

File: src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInStateWithSensitiveAccess.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import org.junit.Test;
+import static io.restassured.RestAssured.given;
+import static org.junit.Assert.assertFalse;
+import gov.ca.cwds.api.FunctionalTest;
+import gov.ca.cwds.api.builder.HttpRequestHandler;
+import gov.ca.cwds.rest.authenticate.UserGroup;
+import gov.ca.cwds.rest.core.Api;
+import io.restassured.http.ContentType;
+
+public class ClientAccessForStaffInStateWithSensitiveAccess extends FunctionalTest {
```
Comment (SergiiRedchuk) [inline src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInStateWithSensitiveAccess.java]: the class name should have been ended with "Test" suffix

File: src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInStateWithSealedAccess.java (hunk 4 above)
Comment (jimlebeau) [reply src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInStateWithSealedAccess.java]: renamed with Test suffix

File: src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInStateWithSensitiveAccess.java (hunk 5 above)
Comment (jimlebeau) [reply src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInStateWithSensitiveAccess.java]: renamed with Test suffix

File: src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInLakeCountyWithSensitiveAccess.java (hunk 3 above)
Comment (jimlebeau) [reply src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInLakeCountyWithSensitiveAccess.java]: renamed with Test suffix

File: src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInLakeCountyWithSealedAccess.java (hunk 2 above)
Comment (jimlebeau) [reply src/test/functional/java/gov/ca/cwds/api/client/access/ClientAccessForStaffInLakeCountyWithSealedAccess.java]: renamed with Test suffix

**Pull request merged**